### PR TITLE
Enable Ctor for DnsEndpointIdentity

### PR DESCRIPTION
**Pull request by hongdai**
* Enable Ctor for DnsEndpointIdentity
* Add unit tests.
* I have verified the unit tests on ProjectN as well.

Fix issue #234

### CODE REVIEW

**Comment by hongdai** [comment]
@dotnet-bot retest this please

**Comment by iamjasonp** [comment]
@dotnet-bot test this please

**Comment by iamjasonp** [comment]
LGTM. 

I can't see why this refuses to pass in CI, but the code looks sound. There will be more PRs in order to enable #12, but this is a good starting point to make sure we test as we go.

File: src/System.ServiceModel.Http/tests/ServiceModel/DnsEndpointEntityTest.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (iamjasonp) [inline src/System.ServiceModel.Http/tests/ServiceModel/DnsEndpointEntityTest.cs]: Is Http the right place to put this test?

**Comment by roncain** [comment]
This PR does not build for me.  Here is an InnerLoop build:
```
                   "c:\OSS\wcf\src\System.ServiceModel.Http\tests\System.ServiceModel.Http.Tests.csproj" (BuildAndTest target) (24) ->
                   (CoreCompile target) -> 
                     ServiceModel\DnsEndpointEntityTest.cs(15,9): error CS0246: The type or namespace name 'DnsEndpointIdentity' could not be found (are you missing a using directive or an assembly reference?) [c:\OSS\wcf\src\System.ServiceModel.Http\tests\System.ServiceModel.Http.Tests.csproj]
                     ServiceModel\DnsEndpointEntityTest.cs(15,53): error CS0246: The type or namespace name 'DnsEndpointIdentity' could not be found (are you missing a using directive or an assembly reference?) [c:\OSS\wcf\src\System.ServiceModel.Http\tests\System.ServiceModel.Http.Tests.csproj]
                     ServiceModel\DnsEndpointEntityTest.cs(25,13): error CS0246: The type or namespace name 'DnsEndpointIdentity' could not be found (are you missing a using directive or an assembly reference?) [c:\OSS\wcf\src\System.ServiceModel.Http\tests\System.ServiceModel.Http.Tests.csproj]
                     ServiceModel\DnsEndpointEntityTest.cs(25,57): error CS0246: The type or namespace name 'DnsEndpointIdentity' could not be found (are you missing a using directive or an assembly reference?) [c:\OSS\wcf\src\System.ServiceModel.Http\tests\System.ServiceModel.Http.Tests.csproj]
```

**Comment by iamjasonp** [comment]
Please dont merge yet. 

The code changes will be dealt with when I fix #9, but I'll keep this PR open for the test code that needs to go in.

File: src/System.ServiceModel.Http/tests/ServiceModel/DnsEndpointEntityTest.cs (hunk 1 above)
Comment (hongdai) [inline src/System.ServiceModel.Http/tests/ServiceModel/DnsEndpointEntityTest.cs]: NetTcp is probably a better place as the E2E scenario we are enabling is a NetTcp scenario. Are there any other suggestions?

**Comment by hongdai** [comment]
@iamjasonp I'll only deal with test code change then.
@roncain I'll investigate and ensure InnerLoop pass on CI machine with the test changes.

**Pull request merged**